File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.py
from collections import deque
'''
        if root==None:
            return []
            
        res=[]
        q=deque([root]) # q=[None]

        while q: 
            level=[]
            for _ in range(len(q)):
                node=q.popleft()
                level.append(node.val)
                if node.left!=None:
                    q.append(node.left)
                if node.right!=None:
                    q.append(node.right)
            res.append(level)
        return res
'''


class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:

        def traversal(root,depth):
            if root==None:
                return

            if depth==len(res):
                res.append([])
            res[depth].append(root.val)
            traversal(root.left,depth+1)
            traversal(root.right,depth+1)
        #----------------------------------------
        res=[]
        traversal(root,0)
        return res

File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public void dfs(TreeNode root, List<List<Integer>> res,int depth){
        if(root==null){
            return;
        }
        if(depth==res.size()){
            res.add(new ArrayList<>());
        }
        res.get(depth).add(root.val);
        dfs(root.left,res,depth+1);
        dfs(root.right,res,depth+1);
    }
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res=new ArrayList<>();
        dfs(root,res,0);
        return res;
    }
}